cart/redis/infrastructure/redisStorage_test.go

```diff
@@ -0,0 +1,412 @@
+				expectedErr: true,
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				got, err := storage.GetCart(context.Background(), tt.key)
+				assert.Equal(t, tt.expectedErr, err != nil)
+				if diff := deep.Equal(got, tt.expected); diff != nil {
+					t.Error("expected response is wrong: ", diff)
+				}
+			})
+		}
+	}
+
+	t.Run("local-redis", func(t *testing.T) {
+		t.Parallel()
+
+		if _, err := exec.LookPath("redis-server"); err != nil {
+			t.Skip("redis-server not installed")
+		}
+
+		server, _ := startUpLocalRedis(t)
+		store := getRedisStorage("unix", server.Socket())
+		runTestCases(t, store)
+	})
+
+	t.Run("docker-redis", func(t *testing.T) {
+		t.Parallel()
+
+		if _, err := exec.LookPath("docker"); err != nil {
+			t.Skip("docker not installed")
+		}
+
+		shutdown, address, _ := startUpDockerRedis(t)
+		defer shutdown()
+		store := getRedisStorage("tcp", address)
+		runTestCases(t, store)
+	})
+}
+
+func TestRedisStorage_HasCart(t *testing.T) {
+	t.Parallel()
+
+	runTestCases := func(t *testing.T, storage *infrastructure.RedisStorage) {
+		t.Helper()
+
+		tests := []struct {
+			name     string
+			key      string
+			expected bool
+		}{
+			{
+				name:     "has existing",
+				key:      existingKey,
+				expected: true,
+			},
+			{
+				name:     "has not non existing",
+				key:      "non",
+				expected: false,
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				got := storage.HasCart(context.Background(), tt.key)
+				assert.Equal(t, tt.expected, got)
+			})
+		}
+	}
+
+	t.Run("local-redis", func(t *testing.T) {
+		t.Parallel()
+
+		if _, err := exec.LookPath("redis-server"); err != nil {
+			t.Skip("redis-server not installed")
+		}
+
+		server, _ := startUpLocalRedis(t)
+		store := getRedisStorage("unix", server.Socket())
+		runTestCases(t, store)
+	})
+
+	t.Run("docker-redis", func(t *testing.T) {
+		t.Parallel()
+
+		if _, err := exec.LookPath("docker"); err != nil {
+			t.Skip("docker not installed")
+		}
+
+		shutdown, address, _ := startUpDockerRedis(t)
+		defer shutdown()
+		store := getRedisStorage("tcp", address)
+		runTestCases(t, store)
+	})
+}
+
+func TestRedisStorage_StoreCart(t *testing.T) {
+	t.Parallel()
+
+	runTestCases := func(t *testing.T, storage *infrastructure.RedisStorage, client redis.UniversalClient) {
+		t.Helper()
+
+		tests := []struct {
+			name  string
+			key   string
+			value *cartDomain.Cart
+		}{
+			{
+				name:  "store new value",
+				key:   "another-test",
+				value: &cartDomain.Cart{ID: "another-test", EntityID: "1"},
+			},
+			{
+				name:  "overwrite existing",
+				key:   "test",
+				value: &cartDomain.Cart{ID: "test", EntityID: "2"},
+			},
+		}
+
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				require.NoError(t, storage.StoreCart(context.Background(), tt.value))
+
+				cmd := client.Get(context.Background(), tt.key)
+				require.NoError(t, cmd.Err())
+
+				result, err := cmd.Bytes()
```

Comment:
please also test here the TTL, and add test data to show difference between Guest/Customer